modules/client/src/internal/client/methods.ts

```diff
@@ -277,76 +279,116 @@ export class ClientMethods extends ClientCore implements IClientMethods {
+              spender: daoAddress,
+              tokenAddress,
+            },
+          );
+        }
       }
-    }
 
-    // Doing the transfer
-    const daoInstance = DAO__factory.connect(daoAddress, signer);
-    const override: { value?: bigint } = {};
+      // Doing the transfer
+      const daoInstance = DAO__factory.connect(daoAddress, signer);
+      const override: { value?: bigint } = {};
 
-    if (tokenAddress === AddressZero) {
-      // Ether
-      override.value = amount;
-    }
+      if (tokenAddress === AddressZero) {
+        // Ether
+        override.value = amount;
+      }
 
-    const tx = await daoInstance.deposit(
-      tokenAddress,
-      amount,
-      reference,
-      override,
-    );
-    yield { key: DaoDepositSteps.DEPOSITING, txHash: tx.hash };
+      const tx = await daoInstance.deposit(
+        tokenAddress,
+        amount,
+        reference,
+        override,
+      );
+      yield { key: DaoDepositSteps.DEPOSITING, txHash: tx.hash };
 
-    const cr = await tx.wait();
-    const log = findLog(cr, daoInstance.interface, "Deposited");
-    if (!log) {
-      throw new FailedDepositError();
-    }
+      const cr = await tx.wait();
+      const log = findLog(cr, daoInstance.interface, "Deposited");
+      if (!log) {
+        throw new FailedDepositError();
+      }
 
-    const daoInterface = DAO__factory.createInterface();
-    const parsedLog = daoInterface.parseLog(log);
+      const daoInterface = DAO__factory.createInterface();
+      const parsedLog = daoInterface.parseLog(log);
 
-    if (!amount.toString() === parsedLog.args["amount"]) {
-      throw new AmountMismatchError(
-        amount,
-        parsedLog.args["amount"].toBigInt(),
+      if (!amount.toString() === parsedLog.args["amount"]) {
+        throw new AmountMismatchError(
+          amount,
+          parsedLog.args["amount"].toBigInt(),
+        );
+      }
+      yield { key: DaoDepositSteps.DONE, amount: amount };
+    } else if (params.type === TokenType.ERC721) {
+      const nftInstance = new Contract(
+        params.tokenAddress,
+        ERC721_ABI,
+        signer,
       );
+
+      // Doing the transfer
+      const tx = await nftInstance["safeTransferFrom(address,address,uint256)"](
+        await signer.getAddress(),
+        params.daoAddressOrEns,
+        params.tokenId,
+      );
+
+      const cr = await tx.wait();
+
+      const log = findLog(cr, nftInstance.interface, "Transfer");
+
+      if (!log) {
+        throw new FailedDepositError();
+      }
+
+      const parsedLog = nftInstance.interface.parseLog(log);
+      if (
+        !parsedLog.args["tokenId"] ||
+        parsedLog.args["tokenId"].toString() !== params.tokenId.toString()
+      ) {
+        throw new FailedDepositError();
+      }
+      yield {
+        key: DaoDepositSteps.DONE,
+        tokenId: params.tokenId,
+      };
+    } else {
+      throw new NotImplementedError("Token type not implemented");
```

Comment:
I would rather mention "invalid"

But the "if" at the top of the function makes this code unreachable. I highly recommend simplifying things